src/components/accordion/Accordion.jsx

```diff
@@ -43,34 +52,34 @@ const Accordion = ({
   className = '',
   spacing = 's',
 }: PropType) => {
-  const [opened, setOpened] = useState({});
+  const [state, dispatch] = useReducer<StateType, ActionType>(reducer, {
+    opened: {},
+  });
 
-  const handleChange = useCallback(
-    (item: string, value: boolean) => {
-      if (allowMultiple) {
-        setOpened({...opened, [item]: value});
-      } else {
-        const newOpened = {...opened};
+  function reducer(state: StateType, action: ActionType): StateType {
```

Comment:
I'd move reducer outside component scope

Reply:
@coderitual ? 😄

Reply:
@bartoszlorek reducer uses component prop so it's easier to keep it that way. Alternatively we could have custom hook with reducer inside and pass prop parameters as hook params but imho this reducer is simple enough to leave it as it is.

Reply:
or have separate actions for a different purposes and via different context dispatches proper one.

Reply:
What do you mean it's harder to test? Reducers should be tested separately. It's implementation detail of component. We should only test public api of component.

Reply:
@bartoszlorek we can jump on a call later if you want so we can discuss that :)

Reply:
@bartoszlorek moreover reducers from useReducer are working more in cheat mode. Please read:

https://twitter.com/dan_abramov/status/1102010979611746304